lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -170,6 +174,94 @@ SessionState::SessionState(
       static_rules_(rule_store),
       credit_map_(4, &ccHash, &ccEqual) {}
 
+/*For 5G which doesn't have response context*/
+SessionState::SessionState(
+    const std::string& imsi, const std::string& session_ctx_id,
+    const SessionConfig& cfg, StaticRuleStore& rule_store)
+    : imsi_(imsi),
+      session_id_(session_ctx_id),
+      // Request number set to 1, because request 0 is INIT call
+      request_number_(1),
+      /*current state would be CREATING and version would be 0 */
+      curr_state_(CREATING),
+      config_(cfg),
+      current_version_(0),
+      static_rules_(rule_store) {}
+
+/* get-set methods of new messages  for 5G*/
+uint32_t SessionState::get_current_version() {
+  return current_version_;
+}
+
+void SessionState::set_current_version(int new_session_version) {
+   current_version_ = new_session_version;
+}
+/* Add PDR rule to this rules session list */
+void SessionState::insert_with_static_rules(SetGroupPDR *rule) {
+  PdrList_.push_back(*rule);
+}
+
+/* Add FAR rule to this rules session list */
+void SessionState::insert_with_static_rules(SetGroupFAR *rule) {
+  FarList_.push_back(*rule);
+}
+
+/* It gets all PDR rule list of the session */
+std::vector<SetGroupPDR>& SessionState::get_5g_static_pdr_rules() {
+  return PdrList_;
+}
+
+/* It gets all FAR rule list of the session */
+std::vector<SetGroupFAR>& SessionState::get_5g_static_far_rules() {
+  return FarList_;
+}
+
+/* UPF client specific retrival rotines, need to revist later */
+magma::lte::Fsm_state_FsmState
+SessionState::SessionInfo::get_smf_state(struct SessionInfo temp) {
+	return temp.state;
+}
+
+std::string   SessionState::SessionInfo::get_sess_id (struct SessionInfo temp) {
+	return temp.sess_id;
+}
+
+uint32_t SessionState::SessionInfo::get_sess_ver_no (struct SessionInfo temp){
+	return temp.ver_no;
+}
+
+NodeId SessionState::SessionInfo::get_node_id (struct SessionInfo temp) {
+	return temp.nodeId;
+}
+
+FSid  SessionState::SessionInfo::getFSeid (struct SessionInfo temp) {
```

Comment:
IS this `get_FSid`, instead of FSeid?

By the way, is `FSid` this subscriber ID?

Reply:
Changed as suggested

Reply:
Changed